### DISCUSSION TITLE
no such network device in the iPXE boot

### DISCUSSION THREAD

**Discussion by gfhoffmann** [ipxe/ipxe#1005] [General]
Hey guys! I'm having the problem below:

![image](https://github.com/ipxe/ipxe/assets/105290508/01676f75-2995-41ba-bafa-41219ee68941)

It seems that iPXE is not finding my network interface, even when it exists on my machine.

How can I solve this problem?

Thanks!

**Comment by NiKiZe** [comment]
You seem to be using a embedded script pointing to net1, the second interface.
Without posting the script you are using we don't know what you are trying to do.

It might be as simple as you should not use any embedded script.
Otherwise add ifstat to the start of your embedded script and post the output.

**Reply by gfhoffmann** [reply]
Thanks for the reply!

My script is like this:

```
#!ipxe

ifopen net{{ INTERFACE_ID }}
set net{{ INTERFACE_ID }}/ip {{ PUBLIC_IP }}
set net{{ INTERFACE_ID }}/netmask 255.255.255.0
set net{{ INTERFACE_ID }}/gateway {{ PUBLIC_GW }}
set net{{ INTERFACE_ID }}/dns 8.8.8.8

chain --autofree https://boot.netboot.xyz
```

We get the INTERFACE_ID, PUBLIC_IP  and PUBLIC_GW programmatically, to use the public interface. In this case, is the interface with id 1.

How to use the ifstat? Can you show me an example using my script above? Because I just tried to insert "ifstat" before the "ifopen" and it didn't change the output.

**Reply by NiKiZe** [reply]
Try adding ifstat before ifopen.
Make sure you rebuild your image and update your boot media.
Which NIC is in the machine? Do you have a pciid? (Use lspci -nn in Linux)

Where does pxelinux come from in this boot sequence?

You might also want to take s look at https://ipxe.org/scripting for ideas on how to deal with errors (I would suggest a || shell at the end of your ifopen line to be able to debug things when it fails.)

**Reply by gfhoffmann** [reply]
I will try to find the answers for your questions, but I was able to put the || shell to debug, this was a great tip! Thanks!

Here you can see the result:

![image](https://github.com/ipxe/ipxe/assets/105290508/73ec41d0-940f-4387-b768-62c675103a4b)

The ifstat returned nothing at all.

**Reply by NiKiZe** [reply]
So you have no NICs in the machine (that iPXE can find)
Most likely no native drivers in iPXE for NICs in the machine.
But we can't help more here without more info.

You are using iPXE in a somewhat advanced way without knowing much about iPXE at all. You might have to do quite s bit of explaining here to get good help.

**Reply by gfhoffmann** [reply]
I'm waiting my colleague to pair with me to try to give you more info about. 
I'm not the expert in the iPXE feature, just trying my best. 
Sorry if I was not able to give all the information right away. 
But I will return with more info to try to understand this problem. 
And thanks for your support so far, its really helping a lot.

**Reply by gfhoffmann** [reply]
Hey there! I'm back with some infos.

> Which NIC is in the machine? Do you have a pciid? (Use lspci -nn in Linux)

c8:00.0 Ethernet controller [0200]: Broadcom Inc. and subsidiaries BCM57416 NetXtreme-E Dual-Media 10G RDMA Ethernet Controller [14e4:16d8] (rev 01)
c8:00.1 Ethernet controller [0200]: Broadcom Inc. and subsidiaries BCM57416 NetXtreme-E Dual-Media 10G RDMA Ethernet Controller [14e4:16d8] (rev 01)

> Where does pxelinux come from in this boot sequence?

In our runtime we get the ipxe code from github, configure everything, run a make  using an EMBED ipxe script that we created to configure the interfaces with proper IPs and chain to other ipxe script.

Just to give some context, we have other 2 machines with the same NIC and everything worked, and in other 2 machines it didn't.

**Reply by NiKiZe** [reply]
Going back then. The screenshot is from a bios booted machine which starts pxelinux, which boots local disk, which then starts iPXE? Why are you not pxebooting into iPXE directly? You should then be able to use undionly.kpxe.

Another thing about net numbers. you should never assume which number a specific nic has. This could change.

**Reply by gfhoffmann** [reply]
Hi there! So, I guess this was the best scenario for our specific use case.
About the net numbers, in the runtime, before booting iPXE, we access the machine and get the updated numbers to get the right one.
